src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -124,6 +125,22 @@ impl QemuInner {
                         network.config.guest_mac.clone().unwrap(),
                     )?;
                 }
+                DeviceType::Protection(prot_dev) => match &prot_dev.config {
+                    ProtectionDeviceConfig::SevSnp(sev_snp_cfg) => {
+                        if sev_snp_cfg.is_snp {
```

Comment:
If my previous idea is ok, I think it will become simple here, just use `cmdline.add_sev_snp_protection_device(...)` without `if .. else ...`